apps/src/code-studio/pd/workshop_dashboard/components/workshop_management.jsx

```diff
@@ -25,9 +26,22 @@ const WorkshopManagement = React.createClass({
     };
   },
 
+  componentWillMount() {
+    if (this.props.showSurveyUrl) {
+      this.permission = new Permission();
+
+      const surveyBaseUrl = this.permission.isOrganizer ? "/organizer_survey_results" : "/survey_results";
+      const surveyUrl = this.props.showSurveyUrl && `${surveyBaseUrl}/${this.props.workshopId}`;
+
+      this.setState({
+        surveyUrl: surveyUrl
+      });
```

Comment:
Correct. I like the fact that `WorkshopManagement` is making the determination here. My point it that for any given instance of WorkshopManagement, the survey URL will never change. It's determined from `permissions` and `props`. There is no reason to store it in the state. Line 34 can simply be `this.surveyUrl = ...` and that's it. Then reference `this.surveyUrl` instead of `this.state.surveyUrl`, like we're doing with `this.permissions`.